### PR TITLE
#5266 [Manager] Remove node should be disabled if there was service on it

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #5266)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/cypress/it/appBar/eventLog.test.ts [hunk 1]
```diff
@@ -22,6 +22,11 @@ describe('App Bar', () => {
   before(() => {
     cy.deleteAllServices();
     cy.createWorkspace({});
+    // Sine we want to do assertion about the event log
+    // make sure event log is already finished
+    cy.findByText('Successfully created workspace workspace1.', {
+      timeout: 70000,
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/appBar/eventLog.test.ts:28]: Nice :)

File: ohara-manager/client/cypress/it/appBar/eventLog.test.ts [hunk 2]
```diff
@@ -22,6 +22,11 @@ describe('App Bar', () => {
   before(() => {
     cy.deleteAllServices();
     cy.createWorkspace({});
+    // Sine we want to do assertion about the event log
+    // make sure event log is already finished
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/appBar/eventLog.test.ts:26]: event log _action_ is

File: ohara-manager/client/cypress/it/appBar/nodeList.test.ts [hunk 3]
```diff
@@ -77,7 +77,11 @@ describe('App Bar', () => {
       // delete the fake node we just added
       cy.findByTestId(`delete-node-${hostname}`).click();
       // confirm dialog
-      cy.findByTestId('confirm-button-DELETE').click();
+      cy.findByTestId('confirm-button-DELETE')
+        // make the click action quick response
+        .click({ timeout: 100 })
+        // the delete button should be disable during removing
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/appBar/nodeList.test.ts:83]: _disabled_ :)

Comment (konekoya) [reply ohara-manager/client/cypress/it/appBar/nodeList.test.ts:83]: And removing -> removal

File: ohara-manager/client/src/components/Layout/NodeListDialog.js [hunk 4]
```diff
@@ -46,6 +48,12 @@ const NodeListDialog = (props) => {
     updateNode(nodeToUpdate);
   };
 
+  const hasUsedServicesInNode = (node) =>
+    !_(node.services)
```
Comment (konekoya) [inline ohara-manager/client/src/components/Layout/NodeListDialog.js:52]: @saivirtue Have you heard lodash FP https://github.com/lodash/lodash/wiki/FP-Guide

It's almost identical to lodash but easier to compose. Maybe we can switch to it if we wish to write our code more like FP style :) 

cc @eechih @wu87988622

File: ohara-manager/client/src/components/Layout/NodeListDialog.js [hunk 5]
```diff
@@ -55,6 +63,13 @@ const NodeListDialog = (props) => {
         onUpdate={handleUpdate}
         options={{
           onRefreshIconClick: fetchNodes,
+          disabledDeleteIcon: hasUsedServicesInNode,
+          deleteTooltip: (node) => {
+            return hasUsedServicesInNode(node)
+              ? 'Cannot remove node which has services in use'
```
Comment (konekoya) [inline ohara-manager/client/src/components/Layout/NodeListDialog.js:69]: Would rephrase this to: _a_ node which has services _running in it_

File: ohara-manager/client/src/components/Layout/NodeListDialog.js (hunk 4 above)
Comment (saivirtue) [reply ohara-manager/client/src/components/Layout/NodeListDialog.js:52]: After some research about `lodash/fp` and read this [fp not convert _.chain()](https://github.com/lodash/lodash/wiki/FP-Guide#chaining) and [use compose instead](https://medium.com/bootstart/why-using-chain-is-a-mistake-9bc1f80d51ba), if the following is more clearly for us, I will vote up for switching to `lodash/fp` 😄 
```
!compose(
  .filter((service) => service.name !== KIND.configurator), 
  .flatMap((service) => service.clusterKeys), 
)(node.services)
.isEmpty();
```

**Comment by saivirtue** [comment]
retry manager-it

**Comment by saivirtue** [comment]
retry manager-it

**Comment by saivirtue** [comment]
retry manager-e2e
